daemon/protocol/python/tests/daemon_cli_test.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python
+
+import os
+import re
+import shutil
+import subprocess
+import tempfile
+import time
+import unittest
+
+gopath = os.environ['GOPATH']
+pypath = gopath + '/src/github.com/scootdev/scoot/daemon/protocol/python/scoot'
+cliRef = ['python', pypath + '/scoot.py']
+createS = ['snapshot', 'create']
+checkoutS = ['snapshot', 'checkout']
+runE = ['exec', 'run']
+pollE = ['exec', 'poll']
+
+echo = ['echo']
+
+class TestCliCommands(unittest.TestCase):
+  daemonProcess = None
+  pypath = ''
+     
+  def setUp(self):
+    self.tmpdir = tempfile.mkdtemp()
+
+    # Note: the following does not work - the process from the pool does not have GOPATH defined so it can't find the binary
+    try:
+      self.daemonProcess = subprocess.Popen(['{0}/bin/daemon'.format(gopath), '-execer_type', 'os'])
+    except Exception as e:
+      self.fail('Fail:{0}'.format(e))  
+    started = False
+    start = time.time()
+    elapsedTime = 0
+    cmd = cliRef + echo + ['ping']
+    while not started and elapsedTime < 4.0:
+      try:    
+        r = subprocess.check_output(cmd, stderr=subprocess.STDOUT)
+        if self.verifyOut(r, 'ping', False):
+          started = True
+      except Exception as e:
+        elapsedTime = time.time() - start
+        time.sleep(0.25)
+       
+    if not started:
+      self.daemonProcess.kill()
+      self.fail("Daemon didn't start within '{0}'".format(elapsedTime))
+
+  def tearDown(self):
+    self.daemonProcess.kill()
+    shutil.rmtree(self.tmpdir)
+    unittest.TestCase.tearDown(self)
+
+  def verifyOut(self, out, expected, failImmediately):
+    m = re.search(expected, out)
+    if failImmediately:
+      self.assertTrue(m != None, "expected to find {0} in '{1}'".format(expected, out))
+           
+    return m != None
+
+  def test_happyPath(self):
+    try:
+      # issue createSnapshot validate that get a snapShot id 
+      cmd = cliRef + createS + [self.tmpdir]
+      r = subprocess.check_output(cmd)
+      self.verifyOut(r, 'snapshot id = [0-9]+', True)
+      m = re.findall(r'[0-9]+', r)
+      sId = m[0]
+      
+      # issue createSnapshot without a path validate get error message and usage prompt 
+      cmd = cliRef + checkoutS + [sId, self.tmpdir]
+      r = subprocess.check_output(cmd).rstrip()
+      self.assertTrue(r == '', "Error: expected '' from checkout, got '{0}'".format(r))
```

Comment:
yes - checkout doesn't display any response to the user (they should be able to see values in the destDir).  We can change this to display a message such as 'Snapshot copied to <destDir>'.  Do we want to do this?